core/src/test/java/com/novoda/merlin/receiver/ConnectivityChangesRegisterShould.java

```diff
@@ -0,0 +1,98 @@
+package com.novoda.merlin.receiver;
+
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.content.IntentFilter;
+import android.net.ConnectivityManager;
+import android.net.NetworkRequest;
+import android.os.Build;
+
+import com.novoda.merlin.service.AndroidVersion;
+import com.novoda.merlin.service.MerlinService;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.*;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+public class ConnectivityChangesRegisterShould {
+
+    @Mock
+    private Context context;
+    @Mock
+    private ConnectivityManager connectivityManager;
+    @Mock
+    private AndroidVersion androidVersion;
+    @Mock
+    private MerlinService merlinService;
+
+    private ConnectivityChangesRegister connectivityChangesRegister;
+
+    @Before
+    public void setUp() {
+        initMocks(this);
+
+        connectivityChangesRegister = new ConnectivityChangesRegister(context, connectivityManager, androidVersion, merlinService);
+    }
+
+    @Test
+    public void givenRegisteredBroadcastReceiverWhenBindingForASecondTimeThenOriginalBroadcastReceieverIsRegisteredAgain() {
```

Comment:
How come these tests are in BDD format (and a bit unreadable as they're missing the `_`s between the clauses) and the others are in the "should" JS-like notation?